### PR TITLE
Remove put_objects line so datasets create correctly

### PR DESCRIPTION
**Pull request by douglas1850**
*Issue #, if available:* N/A

*Description of changes:* Datasets should be not be created using the put_object function; this is resulting in the deletion of dataset "files" deleting the entire dataset itself.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: backend/src/ml_space_lambda/dataset/lambda_functions.py [hunk 1]
```diff
@@ -197,7 +197,6 @@ def create_dataset(event, context):
         raise Exception("Dataset headers do not match expected type and scope.")
 
     if not dataset_dao.get(scope, dataset_name):
-        s3.put_object(Bucket=env_variables["DATA_BUCKET"], Key=directory_name)
```
Comment (dustins) [inline backend/src/ml_space_lambda/dataset/lambda_functions.py:200]: Was this always broken? Why does this cause it to fail, does it just create an exception?

**Comment by douglas1850** [comment]
Looks like I need to update the unit test too to get this to pass the checks, working on it.

File: backend/src/ml_space_lambda/dataset/lambda_functions.py (hunk 1 above)
Comment (douglas1850) [reply backend/src/ml_space_lambda/dataset/lambda_functions.py:200]: This was putting an actual object at the location of the dataset path in the create function. Removing this fixes deleting files from datasets without breaking the creation of datasets.

I confirmed this by running the suite of dataset integtests and they all pass.

**Comment by douglas1850** [comment]
Now it looks like the linter is failing for the unit test file `test_create_dataset`. It's passing with the bb format linter, so I'm not sure why it's failing here...

File: backend/src/ml_space_lambda/dataset/lambda_functions.py (hunk 1 above)
Comment (KyleLilly) [reply backend/src/ml_space_lambda/dataset/lambda_functions.py:200]: This was relatively new. It was added with support for empty datasets but explicitly putting an object with a key of the "directory" resulted in an empty object existing in s3 that matched the dataset path. Normally everything in s3 has a key and there is no actual directories. For the sake of visualization `s3://bucket/my/nested/file` will be rendered as two "directories" `my` and `nested` but there is only one object in the bucket `my/nested/file`. This created an empty object with the key `my/nested/` and then when you added a file we wind up with 2 objects in the bucket `my/nested/` and `my/nested/file`. Having this extra object was the cause of the bug.

Comment (dustins) [reply backend/src/ml_space_lambda/dataset/lambda_functions.py:200]: I see so it wasn't breaking it, it just created an empty file. Thanks!

**Review by dustins — APPROVED** [review]

File: backend/test/dataset/test_create_dataset.py [hunk 2]
```diff
@@ -114,37 +107,9 @@ def test_create_dataset_already_exists(mock_s3, mock_dataset_dao):
     expected_response = generate_html_response(400, error_message)
 
     assert lambda_handler(mock_event, mock_context) == expected_response
-    mock_s3.put_object.assert_not_called()
 
 
-@mock.patch("ml_space_lambda.dataset.lambda_functions.dataset_dao")
-@mock.patch("ml_space_lambda.dataset.lambda_functions.s3")
-def test_create_dataset_client_error(mock_s3, mock_dataset_dao):
```
Comment (KyleLilly) [inline backend/test/dataset/test_create_dataset.py:122]: We _could_ still get throttle from our dynamo during our DAO call but in this particular case I think we have sufficient coverage.

**Review by KyleLilly — APPROVED** [review]

**Pull request merged**